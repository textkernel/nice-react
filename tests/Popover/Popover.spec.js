import React from 'react';
import { Popover } from '../../dist';

const onClick = jest.fn();

describe('<Popover />', () => {
    it('should render a Popover', () => {
        const wrapper = shallow(<Popover
            title="This is a popover"
            controls={ [{
                label: 'OK',
                className: 'text-bold',
                onClick
            }] }
            image="//placeholdit.imgix.net/~text?txtsize=33&txt=Image&w=125&h=100"
            context="info"
            position="bottom right"
        >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. In placerat diam eget tortor facilisis, ut iaculis augue feugiat.
        </Popover>);
        expect(wrapper).toMatchSnapshot();
    });
});
