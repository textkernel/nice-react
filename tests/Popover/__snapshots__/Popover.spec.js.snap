// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover /> should render a Popover 1`] = `
<div
  className="nice-popover info bottom right"
  style={null}
>
  <div
    className="nice-popover-title"
  >
    This is a popover
  </div>
  <div
    className="nice-popover-body"
  >
    <img
      className="nice-popover-image"
      src="//placeholdit.imgix.net/~text?txtsize=33&txt=Image&w=125&h=100"
    />
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. In placerat diam eget tortor facilisis, ut iaculis augue feugiat.
  </div>
  <div
    className="nice-popover-controls"
  >
    <a
      className="nice-popover-control text-bold"
      href="#"
      key="0"
      onClick={[Function]}
    >
      OK
    </a>
  </div>
</div>
`;
